### PR TITLE
Support several reduction ops

### PR DESCRIPTION
**Pull request by mingmingtasd**
Support reduceL1, reduceL2, reduceMax, reduceMean, reduceMin, reduceProduct and reduceSum. PTAL, thanks! @fujunwei

### CODE REVIEW

File: node/src/ops/Reduce.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#ifndef NODE_OPS_REDUCE_H_
+#define NODE_OPS_REDUCE_H_
+
+#include <napi.h>
+#include <webnn/webnn_cpp.h>
+
+namespace node { namespace op {
+
+    enum ReduceOpType {
```
Comment (fujunwei) [inline node/src/ops/Reduce.h:9]: Simplify it as `ReduceType`

File: src/tests/end2end/ReduceTests.cpp [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/tests/WebnnTest.h"
+
+enum ReduceOpType {
```
Comment (fujunwei) [inline src/tests/end2end/ReduceTests.cpp:17]: Move to L.28
```
private:
     enum ReduceType {
        kReduceL1 = 0,
         .....
     }
```

File: src/webnn_native/ops/Reduce.h [hunk 3]
```diff
@@ -12,35 +12,54 @@
...
+#include "common/Log.h"
 #include "webnn_native/Graph.h"
 #include "webnn_native/Operand.h"
-#include "webnn_native/ops/Constant.h"
+#include "webnn_native/Operator.h"
 
 namespace webnn_native { namespace op {
 
-    class ReduceMean final : public OperatorBase {
+    enum ReduceOpType {
```
Comment (fujunwei) [inline src/webnn_native/ops/Reduce.h:25]: Ditto.

File: node/src/ops/Reduce.h (hunk 1 above)
Comment (mingmingtasd) [reply node/src/ops/Reduce.h:9]: Yes, OV hasn't supported now.

File: src/tests/end2end/ReduceTests.cpp (hunk 2 above)
Comment (mingmingtasd) [reply src/tests/end2end/ReduceTests.cpp:17]: CheckReduce() function can't access private ReduceType, how about :
`protected:
     enum ReduceType {
        kReduceL1 = 0,
         .....
     }`

Comment (fujunwei) [reply src/tests/end2end/ReduceTests.cpp]: Ok, please use `protected`

**Comment by mingmingtasd** [comment]
Updated, PTAL, thanks! @fujunwei

**Review by fujunwei — APPROVED** [review]
LGTM.

File: node/src/ops/Reduce.h (hunk 1 above)
Comment (fujunwei) [reply node/src/ops/Reduce.h]: How about `DML` and `NNAPI`, should we report to WebNN Spec?

**Comment by fujunwei** [comment]
Let's merge it, thanks.

**Pull request merged**